### PR TITLE
Add sponsors display on homepage

### PR DESCRIPTION
**Pull request by mosabua**
Currently only confirmed Starburst and GitHub as sponsors. Display is limited to only the logo. No further information or anything.

Needs decision and approval on design (currently three choices implemented), placement, and additional content.

### CODE REVIEW

File: _data/sponsors.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  position: 2
+  anchor: github
+  logo: /assets/images/logos/github.png
+  logosmall: /assets/images/logos/github-small.png
+  logobalanced: /assets/images/logos/github-balanced.png
+  description: |
+    GitHub is used for source code hosting, issue tracking, continuous
+    integration and other aspects of running the project. GitHub is allocating
+    GitHub Actions compute resources to Trino CI beyond the volume available to
+    smaller open source projects.
```
Comment (martint) [inline _data/sponsors.yml:28]: We don't need to include an explanation of how they are providing sponsorship.

File: _data/sponsors.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  links:
+    - urltext: View the GitHub website
+      url: https://www.github.com
+- name: Starburst
+  anchor: starburst
+  position: 1
+  logo: /assets/images/logos/starburst.png
+  logosmall: /assets/images/logos/starburst-small.png
+  logobalanced: /assets/images/logos/starburst-balanced.png
+  description: |
```
Comment (martint) [inline _data/sponsors.yml:38]: Same here

File: _data/sponsors.yml (hunk 1 above)
Comment (mosabua) [reply _data/sponsors.yml:28]: This is not user visible at this stage. Only in the source and can be used later.. should I completely remove it for now?

Comment (mosabua) [reply _data/sponsors.yml]: Removed now.

File: _data/sponsors.yml (hunk 2 above)
Comment (mosabua) [reply _data/sponsors.yml]: Removed now.

**Comment by mosabua** [comment]
All cleaned out and simplified now @martint

**Review by martint — APPROVED** [review]

**Pull request merged**